sdk/search/azure-search-documents/tests/test_search_indexer_client_live.py

```diff
@@ -3,169 +3,173 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
-import json
-from os.path import dirname, join, realpath
-import time
 
-import pytest
-
-from devtools_testutils import AzureMgmtTestCase
-from azure_devtools.scenario_tests import ReplayableTest
-from search_service_preparer import SearchServicePreparer, SearchResourceGroupPreparer
+import random
+import string
 
+import pytest
 from azure.core import MatchConditions
-from azure.core.credentials import AzureKeyCredential
 from azure.core.exceptions import HttpResponseError
-from azure.search.documents.indexes.models import(
-    SearchIndex,
-    SearchIndexerDataSourceConnection,
-    SearchIndexer,
-    SearchIndexerDataContainer,
-)
 from azure.search.documents.indexes import SearchIndexClient, SearchIndexerClient
+from azure.search.documents.indexes.models import (
+    SearchIndex, SearchIndexer, SearchIndexerDataContainer,
+    SearchIndexerDataSourceConnection)
+from devtools_testutils import AzureRecordedTestCase, recorded_by_proxy
+
+from search_service_preparer import SearchEnvVarPreparer, search_decorator
 
-CWD = dirname(realpath(__file__))
-SCHEMA = open(join(CWD, "hotel_schema.json")).read()
-try:
-    BATCH = json.load(open(join(CWD, "hotel_small.json")))
-except UnicodeDecodeError:
-    BATCH = json.load(open(join(CWD, "hotel_small.json"), encoding="utf-8"))
-TIME_TO_SLEEP = 5
-
-class SearchIndexersClientTest(AzureMgmtTestCase):
-    FILTER_HEADERS = ReplayableTest.FILTER_HEADERS + ['api-key']
-
-    def _prepare_indexer(self, endpoint, api_key, name="sample-indexer", ds_name="sample-datasource", id_name="hotels"):
-        con_str = self.settings.AZURE_STORAGE_CONNECTION_STRING
-        self.scrubber.register_name_pair(con_str, 'connection_string')
-        container = SearchIndexerDataContainer(name='searchcontainer')
+
+class TestSearchIndexersClient(AzureRecordedTestCase):
```

Comment:
@mccoyp this is the one test I wrote (the first one) that uses variables. Compare it to the rest of the tests here.